src/shared/components/mutationMapper/MutationMapperStore.ts

```diff
@@ -203,6 +203,27 @@ export default class MutationMapperStore
         }
     }, undefined);
 
+    readonly allTranscripts = remoteData<EnsemblTranscript[] | undefined>({
+        await: () => [
+            this.transcriptsByHugoSymbol,
+            this.canonicalTranscript
+        ],
+        invoke: async()=> {
+            let transcripts: EnsemblTranscript[] = [];
+            if (this.transcriptsByHugoSymbol.result) {
+                transcripts = this.transcriptsByHugoSymbol.result;
+            }
+            // add canonical transcript if it wasn't in response of transcriptsByHugoSymbol
+            if (this.canonicalTranscript.result && !_.some(transcripts, (et => et.transcriptId === this.canonicalTranscript.result!.transcriptId))) {
+                transcripts = transcripts.concat(this.canonicalTranscript.result);
```

Comment:
this kind of logic makes me want a test.  but i think you could maybe compress all 15 lines of this to one line: _.unionBy(this.transcriptsByHugoSymbol.result,[this.cannonicalTrancript.result],(t)=>t.transcriptId)

Reply:
@alisman updated as suggested